From: Ricardo Dias <rdias@suse.com>
Date: Thu, 28 Apr 2016 11:48:05 +0100
Subject: rbd-mirror: Added image-deleter thread to rbd-mirror

Signed-off-by: Ricardo Dias <rdias@suse.com>
(cherry picked from commit 011984d40bce448fe0febfc6a9fcf57d2c7b4768)
(cherry picked from commit 24406c5f8ea12d2c19573116ea84c90ba7446f8b)

Resolves: rhbz#1340482
---
 src/tools/rbd_mirror/Mirror.cc | 10 ++++++++++
 src/tools/rbd_mirror/Mirror.h  |  2 ++
 2 files changed, 12 insertions(+)

diff --git a/src/tools/rbd_mirror/Mirror.cc b/src/tools/rbd_mirror/Mirror.cc
index 9878780..a5bc849 100644
--- a/src/tools/rbd_mirror/Mirror.cc
+++ b/src/tools/rbd_mirror/Mirror.cc
@@ -216,6 +216,9 @@ int Mirror::init()
   // TODO: make interval configurable
   m_local_cluster_watcher.reset(new ClusterWatcher(m_local, m_lock));
 
+  m_image_deleter.reset(new ImageDeleter(m_local, m_threads->timer,
+                                         &m_threads->timer_lock));
+
   return r;
 }
 
@@ -256,6 +259,13 @@ void Mirror::print_status(Formatter *f, stringstream *ss)
 
   if (f) {
     f->close_section();
+    f->open_object_section("image_deleter");
+  }
+
+  m_image_deleter->print_status(f, ss);
+
+  if (f) {
+    f->close_section();
     f->close_section();
     f->flush(*ss);
   }
diff --git a/src/tools/rbd_mirror/Mirror.h b/src/tools/rbd_mirror/Mirror.h
index 298f805..20efe0d 100644
--- a/src/tools/rbd_mirror/Mirror.h
+++ b/src/tools/rbd_mirror/Mirror.h
@@ -14,6 +14,7 @@
 #include "include/rados/librados.hpp"
 #include "ClusterWatcher.h"
 #include "Replayer.h"
+#include "ImageDeleter.h"
 #include "types.h"
 
 namespace rbd {
@@ -59,6 +60,7 @@ private:
   // monitor local cluster for config changes in peers
   std::unique_ptr<ClusterWatcher> m_local_cluster_watcher;
   std::map<peer_t, std::unique_ptr<Replayer> > m_replayers;
+  std::shared_ptr<ImageDeleter> m_image_deleter;
   atomic_t m_stopping;
   bool m_manual_stop = false;
   MirrorAdminSocketHook *m_asok_hook;
